dali/test/python/test_operator_spectrogram.py

```diff
@@ -0,0 +1,227 @@
+
+    def iter_setup(self):
+        data = self.iterator.next()
+        self.feed_input(self.data, data)
+
+def check_operator_spectrogram_vs_python(device, batch_size, input_shape,
+                                         nfft, window_length, window_step):
+    eii1 = RandomDataIterator(batch_size, shape=input_shape, dtype=np.float32)
+    eii2 = RandomDataIterator(batch_size, shape=input_shape, dtype=np.float32)
+    compare_pipelines(
+        SpectrogramPipeline(device, batch_size, iter(eii1), nfft=nfft,
+                            window_length=window_length, window_step=window_step),
+        SpectrogramPythonPipeline(device, batch_size, iter(eii2), nfft=nfft,
+                                  window_length=window_length, window_step=window_step),
+        batch_size=batch_size, N_iterations=5, eps=1e-04)
+
+def test_operator_spectrogram_vs_python():
+    for device in ['cpu']:
+        for batch_size in [3]:
+            for nfft, window_length, window_step, shape in [(256, 256, 128, (1, 4096)),
+                                                            (256, 256, 128, (4096,)),
+                                                            (16, 16, 8, (1, 1000)),
+                                                            (10, 10, 5, (1, 1000)),
+                                                            ]:
+                yield check_operator_spectrogram_vs_python, device, batch_size, shape, \
+                    nfft, window_length, window_step
+
+def check_operator_spectrogram_vs_python_wave_1d(device, batch_size, input_length,
+                                                 nfft, window_length, window_step):
+    f = 4000  # [Hz]
+    sr = 44100  # [Hz]
+    x = np.arange(input_length, dtype=np.float32)
+    y = np.sin(2 * np.pi * f * x / sr)
+
+    data1 = DataIterator(batch_size, y, dtype=np.float32)
+    data2 = DataIterator(batch_size, y, dtype=np.float32)
+
+    compare_pipelines(
+        SpectrogramPipeline(device, batch_size, iter(data1), nfft=nfft,
+                            window_length=window_length, window_step=window_step),
+        SpectrogramPythonPipeline(device, batch_size, iter(data2), nfft=nfft,
+                                  window_length=window_length, window_step=window_step),
+        batch_size=batch_size, N_iterations=5, eps=1e-04)
+
+def test_operator_spectrogram_vs_python_wave():
+    for device in ['cpu']:
+        for batch_size in [3]:
+            for nfft, window_length, window_step, length in [(256, 256, 128, 4096),
+                                                             (16, 16, 8, 1000),
+                                                             (10, 10, 5, 1000),
+                                                             ]:
+                yield check_operator_spectrogram_vs_python_wave_1d, device, batch_size, length, \
+                    nfft, window_length, window_step
+
+def debug_spectrogram(device, batch_size, data1, data2, nfft, window_length, window_step):
```

Comment:
I wonder if we need this debug function included inside this test.

Reply:
removed